lib/fluent/plugin/in_opensearch.rb

```diff
@@ -239,6 +240,43 @@ def parse_time(value, event_time, tag)
       return Time.at(event_time).to_time
     end
 
+    def get_reachable_hosts(hosts=nil)
+      reachable_hosts = []
+      attempt = 0
+      loop do
+        hosts.each do |host|
+          begin
+            if @infinite_check_connection == true
+              check_host = OpenSearch::Client.new(
+                host: ["#{host[:scheme]}://#{host[:host]}:#{host[:port]}"],
+                user: host[:user],
+                password: host[:password],
+                reload_connections: true,
+                resurrect_after: @resurrect_after,
+                reload_on_failure: @reload_on_failure,
+                transport_options: { ssl: { verify: @ssl_verify, ca_file: @ca_file, version: @ssl_version } }
+              )
+              response = check_host.ping  #https://github.com/opensearch-project/opensearch-ruby/blob/136e1c975fc91b8cb80d7d1134e32c6dbefdb3eb/lib/opensearch/api/actions/ping.rb#L33
+              if response == true
+                reachable_hosts << host
+              else
+                log.warn "Connection to #{host[:scheme]}://#{host[:host]}:#{host[:port]} failed with status code #{response.status}"
+              end
+            else
+              reachable_hosts << host
+            end
+          rescue => e
+            log.warn "Failed to connect to #{host[:scheme]}://#{host[:host]}:#{host[:port]}"
+          end
+        end
+        break unless reachable_hosts.empty?
+        log.info "Attempt ##{attempt += 1} to get reachable hosts"
+        log.info "No reachable hosts found. Retrying in #{@request_timeout} seconds..."
+        sleep(@request_timeout)
```

Comment:
Just waiting for periodically is reasonable for this case? In Fluentd, we have [retry_state](https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin_helper/retry_state.rb) plugin helper to create/manage retrying statuses. Could you use it if you have a capability for it?